### PR TITLE
Refactor decolor

### PR DESCRIPTION
**Pull request by shimat**
### This pullrequest changes
- refactored using ReSharper C++
- https://github.com/opencv/opencv/pull/10997#discussion_r172544613
- https://github.com/opencv/opencv/pull/10997#discussion_r172548335
- https://github.com/opencv/opencv/pull/10997#discussion_r172550501

### CODE REVIEW

File: modules/photo/test/test_decolor.cpp [hunk 1]
```diff
@@ -43,8 +43,8 @@
 
 namespace opencv_test { namespace {
 
-TEST(Photo_Decolor, regression)
-{
+    TEST(Photo_Decolor, regression)
+    {
```
Comment (alalek) [inline modules/photo/test/test_decolor.cpp]: No need to add indentation into namespaces.
Please revert changes in this file.

File: modules/photo/src/contrast_preserve.cpp [hunk 2]
```diff
@@ -64,119 +64,92 @@ void cv::decolor(InputArray _src, OutputArray _dst, OutputArray _color_boost)
...
     double E = 0;
     double pre_E = std::numeric_limits<double>::infinity();
 
-    Decolor obj;
-
-    Mat img;
-
-    img = Mat(I.size(),CV_32FC3);
+    Mat img = Mat(I.size(),CV_32FC3);
     I.convertTo(img,CV_32FC3,1.0/255.0);
```
Comment (alalek) [inline modules/photo/src/contrast_preserve.cpp]: `convertTo()` allocates destination image automatically. This is enough:
```.diff
-Mat img = Mat(I.size(),CV_32FC3);
+Mat img;
```

File: modules/photo/src/contrast_preserve.cpp [hunk 3]
```diff
@@ -64,119 +64,92 @@ void cv::decolor(InputArray _src, OutputArray _dst, OutputArray _color_boost)
...
 
-        double pos = 0.0;
-        double neg = 0.0;
-        for(unsigned int i =0;i<alf.size();i++)
+        for(size_t i=0; i<alf.size(); i++)
         {
-            pos = ((1 + alf[i])/2) * exp((-1.0 * 0.5 * pow(temp[i],2))/pow(obj.sigma,2));
-            neg = ((1 - alf[i])/2) * exp((-1.0 * 0.5 * pow(temp1[i],2))/pow(obj.sigma,2));
+            const double pos = ((1 + alf[i])/2) * exp((-1.0 * 0.5 * pow(temp[i],2))/pow(obj.sigma,2));
+            const double neg = ((1 - alf[i])/2) * exp((-1.0 * 0.5 * pow(temp1[i],2))/pow(obj.sigma,2));
```
Comment (alalek) [inline modules/photo/src/contrast_preserve.cpp]: Micro optimization:
```.diff
-pow(someval,2)
+(someval*someval)
```

File: modules/photo/src/contrast_preserve.hpp [hunk 4]
```diff
@@ -81,17 +81,16 @@ int round_num(double a)
     return int(a + 0.5);
```
Comment (alalek) [inline modules/photo/src/contrast_preserve.hpp]: BTW, OpenCV has own optimized `cvRound()` implementation (nearest-to-even).

File: modules/photo/test/test_decolor.cpp (hunk 1 above)
Comment (shimat) [reply modules/photo/test/test_decolor.cpp]: Other tests are indented by 4 spaces but this code is indented by 8 spaces. I decrease indentation of this code. 
https://github.com/opencv/opencv/blob/master/modules/photo/test/test_denoise_tvl1.cpp#L83

**Pull request merged**